webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -1048,6 +1048,45 @@ export function addChannelMember(channelId: string, userId: string, postRootId =
     };
 }
 
+export function addChannelMembers(channelId: string, userIds: string[], postRootId = ''): ActionFuncAsync {
+    const batchSize = 1000;
+    return async (dispatch, getState) => {
+        const channelMembers: ChannelMembership[] = [];
+        try {
+            for (let i = 0; i < userIds.length; i += batchSize) {
+                // eslint-disable-next-line no-await-in-loop
+                const cm = await Client4.addToChannels(userIds.slice(i, i + batchSize), channelId, postRootId);
```

Comment:
This is fine although we could use more parallelism here instead of waiting for each request, thoughts?

Reply:
This is for a server protection. I doubt we really ever hit this limit from our web app.